### PR TITLE
344 345 decompose monoliths

### PR DESCRIPTION
**Pull request by CITguy**
### Basic Rules
0. `src/<component>/<component>.js` -> `src/<component>/<component>.module.js`
  1. only defines the module, its dependencies, and the ngdoc overview
0. `src/<component>/scripts/<item>.[spec|page|exercise].js`
  1. split out from component module file (no matter how small the item)

### JIRA
* [FRMW-344](https://jira.rax.io/browse/FRMW-344)
* [FRMW-345](https://jira.rax.io/browse/FRMW-345)

### LGTMS
- [x] Dev LGTM
- [x] Dev LGTM
- [x] QE LGTM

### CODE REVIEW

File: src/configs/configs.js [hunk 1]
```diff
@@ -1,78 +0,0 @@
...
- *
- * If you do set `customURL` to a non `null` value, then `routesCdnPath.hasCustomURL`
- * will automatically get set to `true`. `hasCustomURL` is intended only for the framework
- * to use, but we are documenting it in case you find your own use case for it.
- *
- * ## Services
- * * {@link configs.service:routesCdnPath routesCdnPath}
- */
-angular.module('encore.ui.configs', [])
-.value('devicePaths', [
```
Comment (CITguy) [inline src/configs/configs.js]: moved to `src/configs/scripts/devicePaths.js`

File: src/configs/configs.js [hunk 2]
```diff
@@ -1,78 +0,0 @@
...
-    { value: '/dev/xvdh', label: '/dev/xvdh' },
-    { value: '/dev/xvdj', label: '/dev/xvdj' },
-    { value: '/dev/xvdk', label: '/dev/xvdk' },
-    { value: '/dev/xvdl', label: '/dev/xvdl' },
-    { value: '/dev/xvdm', label: '/dev/xvdm' },
-    { value: '/dev/xvdn', label: '/dev/xvdn' },
-    { value: '/dev/xvdo', label: '/dev/xvdo' },
-    { value: '/dev/xvdp', label: '/dev/xvdp' }
-])
-.constant('feedbackApi', '/api/encore/feedback')
```
Comment (CITguy) [inline src/configs/configs.js]: moved to `src/configs/scripts/feedbackApi.js`

File: src/configs/configs.js [hunk 3]
```diff
@@ -1,78 +0,0 @@
...
-/**
- * @ngdoc service
- * @name configs.service:routesCdnPath
- * @description [TBD]
- */
-.provider('routesCdnPath', function () {
```
Comment (CITguy) [inline src/configs/configs.js]: moved to `src/configs/scripts/routesCdnPath.js`

File: src/configs/configs.js [hunk 4]
```diff
@@ -1,78 +0,0 @@
...
- * on your configuration.
- *
- * If you do set `customURL` to a non `null` value, then `routesCdnPath.hasCustomURL`
- * will automatically get set to `true`. `hasCustomURL` is intended only for the framework
- * to use, but we are documenting it in case you find your own use case for it.
- *
- * ## Services
- * * {@link configs.service:routesCdnPath routesCdnPath}
- */
-angular.module('encore.ui.configs', [])
```
Comment (CITguy) [inline src/configs/configs.js]: moved to `src/configs/configs.module.js`

File: src/configs/configs.spec.js [hunk 5]
```diff
@@ -1,19 +0,0 @@
...
-        // Inject in angular constructs
-        inject(function (devicePaths) {
-            dp = devicePaths;
-        });
-    });
-
-    it('should exist', function () {
-        expect(dp).to.exist;
-    });
-});
```
Comment (CITguy) [inline src/configs/configs.spec.js]: moved to `src/configs/scripts/devicePaths.spec.js`

File: src/rxApp/rxApp.js [hunk 6]
```diff
@@ -1,1441 +0,0 @@
...
-angular.module('encore.ui.rxApp', [
-    'encore.ui.rxAppRoutes',
-    'encore.ui.rxEnvironment',
-    'ngSanitize',
-    'ngRoute',
-    'cfp.hotkeys',
-    'encore.ui.rxSession',
-    'encore.ui.rxLocalStorage',
-    'encore.ui.rxPermission'
-])
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/rxApp.module.js`

File: src/rxApp/rxApp.js [hunk 7]
```diff
@@ -1,1441 +0,0 @@
...
-/**
- * @ngdoc service
- * @name rxApp.service:encoreRoutes
- * @description
- * Creates a shared instance of AppRoutes that is used for the Encore App nav.
- * This allows apps to make updates to the nav via `encoreRoutes`.
- *
- * @return {Object} Instance of rxAppRoutes with `fetchRoutes` method added
- */
-.factory('encoreRoutes', function (rxAppRoutes, routesCdnPath, rxNotify, $q, $http,
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/encoreRoutes.js`

File: src/rxApp/rxApp.js [hunk 8]
```diff
@@ -1,1441 +0,0 @@
...
- * @param {boolean} [newInstance] Whether the menu items should be a new instance of rxAppRoutes
- * @param {boolean} [hideFeeback] Whether to hide the 'feedback' link or not (defaults to show it)
- * @param {string} [logoutUrl] URL to pass to rx-logout
- *
- * @example
- * <pre>
- * <rx-app site-title="Custom Title"></rx-app>
- * </pre>
- */
-.directive('rxApp', function (encoreRoutes, rxAppRoutes, hotkeys,
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxApp.js`

File: src/rxApp/rxApp.js [hunk 9]
```diff
@@ -1,1441 +0,0 @@
...
- * @param {expression} [title] Title of page
- * @param {expression} [unsafeHtmlTitle] Title for the page, with embedded HTML tags
- * @param {expression} [subtitle] Subtitle of page
- *
- * @example
- * <pre>
- * <rx-page title="'Page Title'"></rx-page>
- * </pre>
- */
-.directive('rxPage', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxPage.js`

File: src/rxApp/rxApp.js [hunk 10]
```diff
@@ -1,1441 +0,0 @@
...
- *
- * @param {object} items Menu items to display. See encoreNav for object definition
- * @param {string} level Level in heirarchy in page. Higher number is deeper nested
- *
- * @example
- * <pre>
- * <rx-app-nav level="1" items="menuItems"></rx-app-nav>
- * </pre>
- */
-.directive('rxAppNav', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxAppNav.js`

File: src/rxApp/rxApp.js [hunk 11]
```diff
@@ -1,1441 +0,0 @@
...
- * @description
- * Creates a menu item. Recursively creates rx-app-nav if 'children' present.
- * 'Item' must be avialable via scope
- *
- * @example
- * <pre>
- * <rx-app-nav-item ng-repeat="item in items"></rx-app-nav-item>
- * </pre>
- */
-.directive('rxAppNavItem', function ($compile, $location, $route) {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxAppNavItem.js`

File: src/rxApp/rxApp.js [hunk 12]
```diff
@@ -1,1441 +0,0 @@
...
- * @restrict E
- * @scope
- * @description
- * Creates a search input form for navigation
- *
- * @param {string} [placeholder] Title of page
- * @param {*} [model] Model to tie input form to (via ng-model)
- * @param {function} [submit] Function to run on submit (model is passed as only argument to function)
- */
-.directive('rxAppSearch', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxAppSearch.js`

File: src/rxApp/rxApp.js [hunk 13]
```diff
@@ -1,1441 +0,0 @@
...
-    };
-})
-/**
- * @ngdoc directive
- * @name rxApp.directive:rxAccountUsers
- * @restrict E
- * @description
- * Provides the ability to switch between account users. This directive is specific to Rackspace
- */
-.directive('rxAccountUsers', function ($location, $route, Encore, $rootScope, encoreRoutes) {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxAccountUsers.js`

File: src/rxApp/rxApp.js [hunk 14]
```diff
@@ -1,1441 +0,0 @@
...
-    };
-})
-/**
- * @ngdoc directive
- * @name rxApp.directive:rxAtlasSearch
- * @restrict E
- * @description
- * Used to search accounts for Cloud Atlas
- */
-.directive('rxAtlasSearch', function ($window) {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxAtlasSearch.js`

File: src/rxApp/rxApp.js [hunk 15]
```diff
@@ -1,1441 +0,0 @@
...
-        }
-    };
-})
-/**
- * @ngdoc directive
- * @name rxApp.directive:rxAccountSearch
- * @restrict E
- * @description [TBD]
- */
-.directive('rxAccountSearch', function ($window) {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxAccountSearch.js`

File: src/rxApp/rxApp.js [hunk 16]
```diff
@@ -1,1441 +0,0 @@
...
-        }
-    };
-})
-/**
- * @ngdoc directive
- * @name rxApp.directive:rxBillingSearch
- * @restrict E
- * @description [TBD]
- */
-.directive('rxBillingSearch', function ($location, $window, encoreRoutes) {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxBillingSearch.js`

File: src/rxApp/rxApp.js [hunk 17]
```diff
@@ -1,1441 +0,0 @@
...
-    };
-})
-/**
- * @ngdoc directive
- * @name rxApp.directive:rxTicketSearch
- * @restrict E
- * @description
- * Used to search tickets for Ticket Queues
- */
-.directive('rxTicketSearch', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxTicketSearch.js`

File: src/rxApp/rxApp.js [hunk 18]
```diff
@@ -1,1441 +0,0 @@
...
- * @ngdoc service
- * @name rxApp.service:rxVisibility
- * @description
- * Provides an interface for adding new `visibility` methods for nav menus.
- * Methods added via `addMethod` should have a `function (scope, args)` interface
- * When you do `visibility: [ "someMethodName", { foo: 1, bar: 2} ]` in
- * a nav menu definition, the (optional) object will be passed to your method as the
- * second argument `args`, i.e. function (scope, args) {}
- */
-.factory('rxVisibility', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxVisibility.js`

File: src/rxApp/rxApp.js [hunk 19]
```diff
@@ -1,1441 +0,0 @@
...
- * @name rxApp.service:rxVisibilityPathParams
- * @description
- * Returns an object with `name` and `method` params that can
- * be passed to `rxVisibility.addMethod()`. We use register this by
- * default, as it's used by the nav menu we keep in routesCdnPath.
- * The method is used to check if {param: 'someParamName'} is present
- * in the current route
- * Use it as `visibility: [ 'rxPathParams', { param: 'userName' } ]`
- */
-.factory('rxVisibilityPathParams', function ($routeParams) {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxVisibilityPathParams.js`

File: src/rxApp/rxApp.js [hunk 20]
```diff
@@ -1,1441 +0,0 @@
...
-/**
- * @ngdoc service
- * @name rxApp.service:rxHideIfUkAccount
- * @description
- * Check if account number in URL is of the UK origin
- *
- * @return {Boolean} false if account number matches UK pattern
- * Use it as `visibility: [ 'rxHideIfUkAccount' ]`
- */
-.factory('rxHideIfUkAccount', function ($routeParams) {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxHideIfUkAccount.js`

File: src/rxApp/rxApp.js [hunk 21]
```diff
@@ -1,1441 +0,0 @@
...
- * @name rxApp.service:rxStatusTags
- * @description
- * This provider is primarily used for applications to specify custom status
- * tags, for use with the `status` attributes of `rx-page` and of breadcrumb
- * objects.
- *
- * It also contains getTag and hasTag run time (vs. config time) methods, but
- * these should rarely, if ever, be needed outside of the framework.
- */
-.provider('rxStatusTags', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxStatusTags.js`

File: src/rxApp/rxApp.js [hunk 22]
```diff
@@ -1,1441 +0,0 @@
...
-/**
-* @ngdoc directive
-* @name rxApp.directive:rxStatusTag
-* @restrict E
-* @scope
-* @description
-* This is used to draw the Alpha/Beta/etc tags in page titles and in breadcrumbs. It's not
-* intended as a public directive.
-*/
-.directive('rxStatusTag', function (rxStatusTags) {
```
Comment (CITguy) [inline src/rxApp/rxApp.js]: moved to `src/rxApp/scripts/rxStatusTag.js`

File: src/rxApp/rxApp.spec.js [hunk 23]
```diff
@@ -1,1156 +0,0 @@
-/* jshint node: true */
-describe('rxApp', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxApp.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 24]
```diff
@@ -1,1156 +0,0 @@
...
-            // get first nav section
-            var navTitle = elCustom[0].querySelector('.nav-section-title');
-
-            // validate it matches custom nav title
-            expect(navTitle.textContent).to.equal(customNav[0].title);
-        });
-    });
-});
-
-describe('rxApp - nav environment detection', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxApp.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 25]
```diff
@@ -1,1156 +0,0 @@
...
-        testEnvironment(mockCdnPath.preprod, 'preprod');
-    });
-
-    it('recognizes the prod environment', function () {
-        mockEnvironment.isUnifiedProd = function () { return true; };
-        testEnvironment(mockCdnPath.production, 'prod');
-    });
-});
-
-describe('rxApp - nav caching', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxApp.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 26]
```diff
@@ -1,1156 +0,0 @@
...
-            expect(mockNotify.add).to.have.been
-                .calledWith(sinon.match('Error'), { type: 'error' });
-            expect(appRoutes.setAll).to.not.have.been.called;
-        });
-
-    });
-
-});
-
-describe('rxApp - customURL', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxApp.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 27]
```diff
@@ -1,1156 +0,0 @@
...
-        expect(isolateScope.isWarning).to.be.true;
-    });
-
-    it('should set the local nav file warning message', function () {
-        expect(isolateScope.warningMessage).to.contain('You are using a local nav file');
-    });
-
-});
-
-describe('rxApp - preprod environment', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxApp.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 28]
```diff
@@ -1,1156 +0,0 @@
...
-    });
-
-    it('should set the preprod warning message', function () {
-        expect(isolateScope.warningMessage)
-            .to.contain('You are using a pre-production environment that has real, live production data!');
-    });
-
-});
-
-describe('rxAppNav', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxAppNav.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 29]
```diff
@@ -1,1156 +0,0 @@
...
-    });
-
-    it('should add "level" class of appropriate level', function () {
-        expect(el.hasClass('rx-app-nav-level-1')).to.be.true;
-    });
-
-//    TODO allow children to be dynamically injected
-});
-
-describe('rxAppNavItem', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxAppNavItem.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 30]
```diff
@@ -1,1156 +0,0 @@
...
-
-        it('should not when when not all roles are present', function () {
-            var item = el.find('a:contains("multipleAllRolesFailure")').parent();
-            expect(item.hasClass('ng-hide')).to.be.true;
-        });
-    });
-
-});
-
-describe('rxPage', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxPage.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 31]
```diff
@@ -1,1156 +0,0 @@
...
-    });
-
-    it('should update page title if title changes', function () {
-        scope.myCustomTitle = 'abc';
-        scope.$digest();
-        expect(pageTitle.getTitle()).to.equal('abc');
-    });
-});
-
-describe('rxVisibility', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxVisibility.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 32]
```diff
@@ -1,1156 +0,0 @@
...
-        expect(rxvisibility.getMethod('foo'), 'getMethod').to.equal(method);
-
-    });
-
-    it('should return undefined for an unknown method', function () {
-        expect(rxvisibility.getMethod('foo'), 'getMethod').to.be.undefined;
-    });
-});
-
-describe('rxVisibilityPathParams', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxVisibilityPathParams.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 33]
```diff
@@ -1,1156 +0,0 @@
...
-            var scope = $rootScope.$new();
-            var method = rxVisibilityPathParams.method;
-            expect(method(scope, { param: 'barId' }), ':barId should be present').to.be.true;
-            expect(method(scope, { param: 'abc' }), ':abc is not defined, should not be present').to.be.false;
-        });
-
-    });
-});
-
-describe('rxHideIfUkAccount', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxHideIfUkAccount.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 34]
```diff
@@ -1,1156 +0,0 @@
...
-    it('should return true if not UK account', function () {
-        $location.path('/accounts/9999999');
-        $rootScope.$digest();
-        expect($routeParams).to.deep.equal({ accountNumber: '9999999' });
-        var method = rxHideIfUkAccount.method;
-        expect(method(), 'Non-UK Account should return true').to.be.true;
-    });
-});
-
-describe('rxAccountUsers', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxAccountUsers.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 35]
```diff
@@ -1,1156 +0,0 @@
...
-        userSelect.remove();
-        expect(unregisterCheckCloud.called).to.eq(true);
-    });
-
-    afterEach(function () {
-        rootScopeStub.restore();
-    });
-});
-
-describe('rxStatusTags - custom status tags', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxStatusTags.spec.js`

File: src/rxApp/rxApp.spec.js [hunk 36]
```diff
@@ -1,1156 +0,0 @@
...
-        var config = rxstatusTags.getTag('missingKey');
-        expect(config.text).to.equal('');
-        expect(config.class).to.equal('');
-        expect(rxstatusTags.hasTag('missingKey')).to.be.false;
-
-    });
-
-});
-
-describe('rxStatusTag', function () {
```
Comment (CITguy) [inline src/rxApp/rxApp.spec.js]: moved to `src/rxApp/scripts/rxStatusTag.spec.js`

File: src/rxApp/scripts/rxApp.page.js [hunk 37]
```diff
@@ -104,63 +102,3 @@ exports.rxApp = {
         return Page.create(rxApp);
     }
 };
-
-var rxPage = {
```
Comment (CITguy) [inline src/rxApp/scripts/rxApp.page.js]: moved to `src/rxApp/scripts/rxPage.page.js`

File: src/rxApp/scripts/rxApp.page.js [hunk 38]
```diff
@@ -104,63 +102,3 @@ exports.rxApp = {
...
-        get: function () {
-            var page = this;
-            return this.lblTitleTag.isPresent().then(function (present) {
-                return present ? page.lblTitleTag.getText() : null;
-            });
-        }
-    }
-};
-
-exports.rxPage = {
```
Comment (CITguy) [inline src/rxApp/scripts/rxApp.page.js]: moved to `src/rxApp/scripts/rxPage.page.js`

**Comment by bry4n** [comment]
LGTM. :+1: reduce monolith files

**Comment by 100stacks** [comment]
LGTM. Nice work :+1:

**Comment by CITguy** [comment]
@Droogans, do you have any comments on this PR. I included QE LGTM to keep you informed of new file locations.

**Comment by Droogans** [comment]
@CITguy might want to doublecheck that the page object node module will generate correctly.

I think [this glob](https://github.com/rackerlabs/encore-ui/blob/81c4ac500701d678afd2997602c6115733bc5338/grunt-tasks/options/concat.js#L41) that's used to concat all page object files together needs an extra star or two in there.

**Comment by CITguy** [comment]
The file glob was [updated](https://github.com/rackerlabs/encore-ui/pull/1334/files#diff-857e53ba4cbe9e089f1049c5ea22e03fR41) to add required asterisks.

The point of this PR (and upcoming PRs) is to split up the monolithic source files to make it easier to move source code around for Atoms/Molecules support coming up. While some files may look tiny, this is to ensure that all relevant sources are appropriately split from the monolith.

**Comment by CITguy** [comment]
@jj4th, any comments?

**Comment by jj4th** [comment]
I can't find anything I disagree with here.

**Pull request merged**